internal/provider/resource/dynamic_secret/dynamic_secret_kubernetes.go

```diff
@@ -0,0 +1,424 @@
+				Attributes: map[string]schema.Attribute{
+					"service_account_name": schema.StringAttribute{
+						Required:    true,
+						Description: "Name of the service account to generate tokens for.",
+					},
+					"namespace": schema.StringAttribute{
+						Required:    true,
+						Description: "Kubernetes namespace where the service account exists.",
+					},
+				},
+			},
+			"dynamic_config": schema.SingleNestedAttribute{
+				Optional:    true,
+				Description: "Configuration for the 'dynamic' credential type.",
+				Attributes: map[string]schema.Attribute{
+					"allowed_namespaces": schema.StringAttribute{
+						Required:    true,
+						Description: "Kubernetes namespace(s) where the service accounts will be created. You can specify multiple namespaces as a comma-separated list (e.g., “default,kube-system”). During lease creation, you can specify which namespace to use from this allowed list.",
+					},
+					"role": schema.StringAttribute{
+						Required:    true,
+						Description: "Name of the role to assign to the temporary service account.",
+					},
+					"role_type": schema.StringAttribute{
+						Required:    true,
+						Description: "Type of role to assign ('cluster-role' or 'role').",
+					},
+				},
+			},
+
+			"audiences": schema.ListAttribute{
+				Optional:    true,
+				Description: "Optional list of audiences to include in the generated token.",
+				ElementType: types.StringType,
+			},
+		},
+
+		ReadConfigurationFromPlan: func(ctx context.Context, plan DynamicSecretBaseResourceModel) (map[string]interface{}, diag.Diagnostics) {
+			configurationMap := make(map[string]interface{})
+			var configuration DynamicSecretKubernetesConfigurationModel
+
+			diags := plan.Configuration.As(ctx, &configuration, basetypes.ObjectAsOptions{})
+			if diags.HasError() {
+				return nil, diags
+			}
+
+			if !configuration.GatewayId.IsNull() && !configuration.GatewayId.IsUnknown() {
+				configurationMap["gatewayId"] = configuration.GatewayId.ValueString()
+			}
+
+			configurationMap["authMethod"] = configuration.AuthMethod.ValueString()
+
+			switch configuration.AuthMethod.ValueString() {
+			case "api":
+				if configuration.ApiConfig != nil {
+					configurationMap["url"] = configuration.ApiConfig.ClusterUrl.ValueString()
+					configurationMap["clusterToken"] = configuration.ApiConfig.ClusterToken.ValueString()
+
+					if !configuration.ApiConfig.EnableSsl.IsNull() && !configuration.ApiConfig.EnableSsl.IsUnknown() {
+						configurationMap["sslEnabled"] = configuration.ApiConfig.EnableSsl.ValueBool()
+					}
+					if !configuration.ApiConfig.Ca.IsNull() && !configuration.ApiConfig.Ca.IsUnknown() {
+						configurationMap["ca"] = configuration.ApiConfig.Ca.ValueString()
+					}
+				} else {
+					diags.AddError(
+						"Invalid Configuration",
+						"When auth_method is 'api', the 'api_config' block must be provided.",
+					)
+					return nil, diags
+				}
+			case "gateway":
```

Comment:
let's add proper handling for this one please!
<img width="982" alt="image" src="https://github.com/user-attachments/assets/1c4e4bcb-f6d0-4c97-90fd-272636853e6d" />